usaspending_api/awards/v2/views/accounts.py

```diff
@@ -0,0 +1,121 @@
+from collections import OrderedDict
+
+from psycopg2.sql import Identifier, Literal, SQL
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.helpers.generic_helper import get_pagination
+from usaspending_api.common.helpers.sql_helpers import execute_sql_to_ordered_dictionary
+from usaspending_api.common.validator.award import get_internal_or_generated_award_id_model
+from usaspending_api.common.validator.pagination import customize_pagination_with_sort_columns
+from usaspending_api.common.validator.tinyshield import validate_post_request
+
+
+SORTABLE_COLUMNS = {
+    "federal_account": "federal_account",
+    "total_transaction_obligated_amount": "total_transaction_obligated_amount",
+    "agency": "funding_agency_name",
+    "account_title": "fa.account_title",
+}
+
+
+DEFAULT_SORT_COLUMN = "federal_account"
+
+
+ACCOUNTS_SQL = SQL(
+    """
+    with gather_financial_accounts_by_awards as (
+        select  a.funding_agency_id,
+                nullif(faba.transaction_obligated_amount, 'NaN') transaction_obligated_amount,
+                faba.treasury_account_id
+        from    awards a
+                inner join financial_accounts_by_awards faba on faba.award_id = a.id
+        where   {award_id_column} = {award_id}
+    ), agency_id_to_agency_id_for_toptier_mapping as (
+        select
+            a.id                            agency_id,
+            t.agency_id                     agency_id_for_toptier,
+            t.toptier_agency_name,
+            t.toptier_agency_abbreviation
+        from (
+                select
+                    a.id                    agency_id,
+                    ta.toptier_agency_id,
+                    ta.name                 toptier_agency_name,
+                    ta.abbreviation         toptier_agency_abbreviation,
+                    row_number() over(
+                        partition by ta.toptier_agency_id
+                        order by sa.name is not distinct from ta.name desc, a.update_date asc, a.id desc
+                    ) as per_toptier_row_number
+                from
+                    agency a
+                    inner join toptier_agency ta on ta.toptier_agency_id = a.toptier_agency_id
+                    left outer join subtier_agency sa on sa.subtier_agency_id = a.subtier_agency_id
+            ) t
+            inner join agency a on a.toptier_agency_id = t.toptier_agency_id
+        where
+            t.per_toptier_row_number = 1
+    )
+    select
+        sum(gfaba.transaction_obligated_amount)         total_transaction_obligated_amount,
```

Comment:
Does this not need to be coalesced to 0.0 like funding_rollup?

Reply:
Took that directly from the IDV implementation and overlooked that. Don't see why it shouldn't be added. Can throw that on similar to funding_rollup.